EIPS/eip-7017.md

```diff
@@ -0,0 +1,223 @@
+- An NFT marketplace informing an NFT owner about an offer being made to one of it’s NFTs.
+- A metaverse informing about an important event.
+- Warning to an address about an ENS domain expiration date approaching.
+
+## Motivation
+
+With the adoption of web3 applications, an increasing necessity arises to be informed about certain events happening on-chain.
+
+Users are used to being informed, whether it be about news or updates of their favorite applications. As we are in a time of instant data feeds on social media, instant messaging apps and notifications of events that users care about, notifications are a feature in practically every application that exists in web2. They mostly come to email inboxes, SMSs, inside the applications, or in the notification inbox in the operating system.
+
+If they would be taken away, the engagement on these web2 applications would sink. This is not different with web3 applications: users cannot be left in the dark about what is going on in them. Not only that, for some applications, all that matters is the participation of users on certain events, like governance on a DAO.
+
+In web2, most of the user account’s are linked to an email address that is required at sign-up. This makes it easy to send notifications to specific users and requires no further complexity as an infrastructure exists to deliver a message to the user using their email address.
+
+In web3, on the other hand, there's mostly is only one inbox to send notifications to: addresses.
+Every smart contract can define its own event’s to which one can listen to, but for each of them, a change has to be done in the frontend to listen to that specific contract and event structure. This poses a problem of coordination between smart contracts and web3 applications that can notify users.
+
+This EIP aims at proposing a decentralized approach to send and receive notifications from and to ethereum addresses, including smart contracts, in a standarized way, facilitating front-end intrgrations. These could be:
+
+- Wallets
+- Mobile phone dapps
+- Email services
+- Discord / Telegram Bots
+
+All of these could be notifying about on-chain notifications to the user.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+An approach that is both simple, decentralized and easy to implement is to use a notifications smart contract standard that is **event-based** to be able to emit notifications to one address or broadcast them to anyone that wants to listen. Off-chain **whitelists** would record all addresses a user wants to allow receiving messages from to avoid spam. This is useful for direct messages from one address to another. Off-chain **subscription lists** would indicate which addresses they want to listen for general broadcasts. This is useful for receiving updates on a project.
```

Comment:
This paragraph introduces no requirements or recommendations for implementation. It should be split and moved to motivation and rationale as appropriate.